src/app/parcels/form/ParcelForm.tsx

```diff
@@ -174,6 +189,47 @@ const ParcelForm: React.FC<ParcelFormProps> = ({
         }
     }, [clientDetails, clientIdForFetch]);
 
+    useEffect(() => {
+        const getTimeSlots = async (): Promise<void> => {
+            if (fields.collectionCentre) {
+                const { data, error } = await getActiveTimeSlotsForCollectionCentre(
+                    fields.collectionCentre,
+                    supabase
+                );
+
+                if (error) {
+                    let errorMessage;
+                    switch (error.type) {
+                        case "collectionTimeSlotsFetchFailed":
+                            errorMessage = "Failed to fetch collection time slots";
+                            break;
+                        case "noCollectionTimeSlotsData":
+                            errorMessage = "No collection time slots data available";
+                            break;
+                    }
+                    setSubmitErrorMessage(`${errorMessage}. Log ID: ${error.logId}`);
+                    return;
+                }
+
+                if (data.length === 0) {
+                    setFormErrors({
+                        ...initialFormErrors,
+                        collectionSlot: Errors.noCollectionSlotsSet,
+                    });
+                } else {
+                    setFormErrors({
+                        ...initialFormErrors,
+                        collectionSlot: Errors.initial,
+                    });
```

Comment:
Does this one show up if you press submit without selecting a timeslot?

Reply:
Yes, if there are no collection slots for the particular collection centre then an error displays: "There are no collection slots set for this collection centre, please select a different collection centre or contact admin." on the card but once a collection centre with time slot data is selected then it resets the error to "required".